napari/_qt/dialogs/_tests/test_qt_plugin_dialog.py

```diff
@@ -149,3 +270,106 @@ def test_constructor_visible_widgets(plugin_dialog_constructor):
+        plugin_dialog.installed_list.handle_action(
+            item,
+            'test-name-1',
+            InstallerActions.UPGRADE,
+        )
+        mock.assert_called_with(
+            trans._("updating..."), InstallerActions.UPGRADE
+        )
+
+    with patch.object(qt_plugin_dialog.WarnPopup, "exec_") as mock:
+        plugin_dialog.installed_list.handle_action(
+            item,
+            'test-name-1',
+            InstallerActions.UNINSTALL,
+        )
+        assert mock.called
+
+    item = plugin_dialog.available_list.item(0)
+    with patch.object(qt_plugin_dialog.PluginListItem, "set_busy") as mock:
+        plugin_dialog.available_list.handle_action(
+            item,
+            'test-name-1',
+            InstallerActions.INSTALL,
+            version='3',
+        )
+        mock.assert_called_with(
+            trans._("installing..."), InstallerActions.INSTALL
+        )
+
+        plugin_dialog.available_list.handle_action(
+            item, 'test-name-1', InstallerActions.CANCEL, version='3'
+        )
+        mock.assert_called_with(
+            trans._("cancelling..."), InstallerActions.CANCEL
+        )
+
+    # Wait for refresh timer, state and worker to be done
+    qtbot.waitUntil(
+        lambda: not plugin_dialog._add_items_timer.isActive()
+        and plugin_dialog.refresh_state == qt_plugin_dialog.RefreshState.DONE
+    )
+    qtbot.waitUntil(lambda: not plugin_dialog.worker.is_running)
+
+
+def test_on_enabled_checkbox(plugin_dialog, qtbot, plugins, old_plugins):
+    # checks npe2 lines
+    item = plugin_dialog.installed_list.item(0)
+    widget = plugin_dialog.installed_list.itemWidget(item)
+
+    assert plugins.plugins['my-plugin'] is True
+    with qtbot.waitSignal(widget.enabled_checkbox.stateChanged, timeout=500):
+        widget.enabled_checkbox.setChecked(False)
+    assert plugins.plugins['my-plugin'] is False
+
+    # checks npe1 lines
+    item = plugin_dialog.installed_list.item(1)
+    widget = plugin_dialog.installed_list.itemWidget(item)
+
+    assert old_plugins.enabled[0] is True
+    with qtbot.waitSignal(widget.enabled_checkbox.stateChanged, timeout=500):
+        widget.enabled_checkbox.setChecked(False)
+    assert old_plugins.enabled[0] is False
+
+
+def test_add_items_outdated(plugin_dialog):
+    """Test that a plugin is tagged as outdated (a newer version is available), the update button becomes visible."""
+    new_plugin = (
+        npe2.PackageMetadata(name="my-plugin", version="0.4.0"),
+        True,
+        {
+            "home_page": 'www.mywebsite.com',
+            "pypi_versions": ['0.3.0'],
+            "conda_versions": ['0.3.0'],
+        },
+    )
+
+    plugin_dialog._plugin_data = [new_plugin]
+
+    plugin_dialog._add_items()
+    item = plugin_dialog.installed_list.item(0)
+    widget = plugin_dialog.installed_list.itemWidget(item)
+
+    assert widget.update_btn.isVisible()
```

Comment:
Why update button is visible if the installed version is newer than the available version?

Reply:
sorry, this is confusing I realize now.  

I am adding this plugin to the list of available plugins. When the dialog is being built this one will be listed as available, and it will be found as already installed.  Then, it will check if the installed version is a lower version than the one available.  In this case, my-plugin is installed with version 0.1.0, so the one we are trying to install is newer (v 0.4.0), so the update button should pop up.

I added this note to the comments.  I also realized the versions available in conda and PyPI should look less confusing.  They are actually not used in the check for the latest version.  I still updated it in the test.

Hope this makes sense.